### ISSUE TITLE
In-App Messaging autolayout from background thread crash

### ISSUE THREAD

**Issue by fr-machlin** [firebase/firebase-ios-sdk#4124]
### [READ] Step 1: Are you in the right place?
YES

### [REQUIRED] Step 2: Describe your environment

  * Xcode version: 11.1
  * Firebase SDK version: 6.10
  * Firebase Component: InAppMessaging
  * Component version: I'm using the Carthage version 6.10 not sure about component version.

#### Steps to reproduce:

I created and deployed my first In-App Message dialog. When I tested the campaign on my device I started to experience crash shortly after the app launches.

Nothing in the stack trace below is from my code, it's all from the Firebase SDK.

#### Relevant Code:

No client side code

#### Stack Trace:

```

2019-10-20 00:44:25.533320+0800 Circle[57637:1636117] 6.10.0 - [Firebase/InAppMessaging][I-IAM900014] No fetch epoch time detected in API response.
2019-10-20 00:44:25.533468+0800 Circle[57637:1636117] 6.10.0 - [Firebase/InAppMessaging][I-IAM900008] A test message with id 5459849288093794304 was parsed successfully.
2019-10-20 00:44:25.533634+0800 Circle[57637:1636117] 6.10.0 - [Firebase/InAppMessaging][I-IAM900002] 2 message definitions were parsed out successfully and 0 messages are discarded
2019-10-20 00:44:25.533756+0800 Circle[57637:1636117] 6.10.0 - [Firebase/InAppMessaging][I-IAM130012] API request for fetching messages and parsing the response was successful.
2019-10-20 00:44:25.533955+0800 Circle[57637:1636117] 6.10.0 - [Firebase/InAppMessaging][I-IAM700004] 2 messages were fetched successfully.
2019-10-20 00:44:25.534052+0800 Circle[57637:1636117] 6.10.0 - [Firebase/InAppMessaging][I-IAM270005] No impression records update due to no change after applying the server message list
2019-10-20 00:44:25.534180+0800 Circle[57637:1636117] 6.10.0 - [Firebase/InAppMessaging][I-IAM150004] Persistent file path for fetch response data is /Users/01143870/Library/Developer/CoreSimulator/Devices/9F38FFAA-FA6C-47C2-8DEE-DCA229D6B410/data/Containers/Data/Application/0827E197-C6C9-4646-BB73-F42F896613AB/Library/Caches/firebase-iam-messages-cache
2019-10-20 00:44:25.549062+0800 Circle[57637:1636117] 6.10.0 - [Firebase/InAppMessaging][I-IAM160010] There are analytics event trigger based messages, enable listening
2019-10-20 00:44:25.549185+0800 Circle[57637:1636117] 6.10.0 - [Firebase/InAppMessaging][I-IAM140002] Start observing Firebase Analytics events for rendering messages.
2019-10-20 00:44:25.549289+0800 Circle[57637:1636117] 6.10.0 - [Firebase/InAppMessaging][I-IAM160001] There are 1 test messages and 1 regular messages and 2 Firebase Analytics events to watch after resetting the message cache
2019-10-20 00:44:25.549357+0800 Circle[57637:1636117] 6.10.0 - [Firebase/InAppMessaging][I-IAM240002] Fetch is done. Start message rendering flow.
2019-10-20 00:44:26.082667+0800 Circle[57637:1636869] 6.10.0 - [Firebase/InAppMessaging][I-IAM160007] Returning a test message for app foreground display
2019-10-20 00:44:26.334395+0800 Circle[57637:1635929] 6.10.0 - [Firebase/Analytics][I-ACS002001] Measurement timer fired
2019-10-20 00:44:26.334621+0800 Circle[57637:1635929] 6.10.0 - [Firebase/Analytics][I-ACS002003] Measurement timer canceled
2019-10-20 00:44:26.334826+0800 Circle[57637:1635929] 6.10.0 - [Firebase/Analytics][I-ACS023033] Starting data upload
2019-10-20 00:44:26.335690+0800 Circle[57637:1635929] 6.10.0 - [Firebase/Analytics][I-ACS023105] Event is not subject to real-time event count daily limit. Marking an event as real-time. Event name, parameters: display_screen, {
    action = display_screen;
    condition = location=global&gender=all&recency=alltime;
    displayed_at = 2019/10/20 00:44:25 +0800;
    ga_debug (_dbg) = 1;
    ga_event_origin (_o) = app;
    ga_realtime (_r) = 1;
    screen_name = Explore;
    source = background;
}
2019-10-20 00:44:26.336885+0800 Circle[57637:1635929] 6.10.0 - [Firebase/Analytics][I-ACS012018] Saving bundle. size (bytes): 566
2019-10-20 00:44:26.337065+0800 Circle[57637:1635929] 6.10.0 - [Firebase/Analytics][I-ACS023116] Bundle added to the upload queue. BundleID, timestamp (ms): 321, 1571503465306
2019-10-20 00:44:26.338647+0800 Circle[57637:1636869] 6.10.0 - [Firebase/Analytics][I-ACS023038] Uploading events. Elapsed time since last successful upload (s): 2.299102067947388
2019-10-20 00:44:26.338759+0800 Circle[57637:1636869] 6.10.0 - [Firebase/Analytics][I-ACS023039] Measurement data sent to network. Timestamp (ms), data: 1571503466338, <APMPBMeasurementBatch: 0x600002b3efd0>
2019-10-20 00:44:26.341859+0800 Circle[57637:1636869] 6.10.0 - [Firebase/Analytics][I-ACS900000] Uploading data. Host: https://app-measurement.com/a
2019-10-20 00:44:26.780083+0800 Circle[57637:1636867] 6.10.0 - [Firebase/Analytics][I-ACS901006] Received SSL challenge for host. Host: https://app-measurement.com/a
2019-10-20 00:44:26.869967+0800 Circle[57637:1636870] 6.10.0 - [Firebase/InAppMessaging][I-IAM000004] No image MIME type text/html detected for URL https://www.uniqlo.com/jp/store/goods/419285-67
2019-10-20 00:44:26.870311+0800 Circle[57637:1636870] 6.10.0 - [Firebase/InAppMessaging][I-IAM400019] Error in loading image data for the message.
2019-10-20 00:44:26.870630+0800 Circle[57637:1636870] 6.10.0 - [Firebase/InAppMessaging][I-IAM400009] Display ran into error for message 5459849288093794304: Error Domain=com.firebase.inappmessaging Code=1 "No image MIME type text/html detected for URL https://www.uniqlo.com/jp/store/goods/419285-67" UserInfo={NSLocalizedDescription=No image MIME type text/html detected for URL https://www.uniqlo.com/jp/store/goods/419285-67}
2019-10-20 00:44:26.870986+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.871210+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.871379+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.871576+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.871754+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.871921+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.891412+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.891544+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.891689+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.891915+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.892087+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.892487+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.892636+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.899940+0800 Circle[57637:1636868] [Animation] +[UIView setAnimationsEnabled:] being called from a background thread. Performing any operation from a background thread on UIView or a subclass is not supported and may result in unexpected and insidious behavior. trace=(
	0   UIKitCore                           0x00007fff47a44b5d kFixedAnimationDuration_block_invoke_3 + 119
	1   libdispatch.dylib                   0x000000010c0e7d6c _dispatch_client_callout + 8
	2   libdispatch.dylib                   0x000000010c0e92bb _dispatch_once_callout + 66
	3   UIKitCore                           0x00007fff47a44ae4 +[UIView(Animation) setAnimationsEnabled:] + 62
	4   UIKitCore                           0x00007fff47a44c38 +[UIView(Animation) performWithoutAnimation:] + 78
	5   UIKitCore                           0x00007fff47959081 -[UIStackView _commonStackViewInitializationWithArrangedSubviews:] + 122
	6   UIKitCore                           0x00007fff47958fcf -[UIStackView initWithFrame:] + 103
	7   UIKitCore                           0x00007fff46b0c711 -[_UIInterfaceActionSeparatableSequenceView initWithVisualStyle:] + 351
	8   UIKitCore                           0x00007fff46b0bb74 -[_UIInterfaceActionRepresentationsSequenceView initWithVisualStyle:] + 328
	9   UIKitCore                           0x00007fff46afa7f7 -[UIInterfaceActionGroupView _loadActionSequenceView] + 79
	10  UIKitCore                           0x00007fff46af7efe -[UIInterfaceActionGroupView _commonInitWithActionGroup:visualStyleProvider:actionHandlerInvocationDelegate:] + 522
	11  UIKitCore                           0x00007fff46af7fd3 -[UIInterfaceActionGroupView initWithActionGroup:actionHandlerInvocationDelegate:] + 138
	12  UIKitCore                           0x00007fff46b2341f -[_UIAlertControllerInterfaceActionGroupView initWithAlertController:actionGroup:actionHandlerInvocationDelegate:] + 87
	13  UIKitCore                           0x00007fff46b43dd1 -[_UIAlertControllerView _prepareMainInterfaceActionsGroupView] + 150
	14  UIKitCore                           0x00007fff46b430d6 -[_UIAlertControllerView _prepareViewsAndAddConstraints] + 40
	15  UIKitCore                           0x00007fff46b43071 -[_UIAlertControllerView setAlertController:] + 93
	16  UIKitCore                           0x00007fff46b2c572 -[UIAlertController loadView] + 81
	17  UIKitCore                           0x00007fff46f0895f -[UIViewController loadViewIfRequired] + 172
	18  UIKitCore                           0x00007fff46f0910c -[UIViewController view] + 27
	19  UIKitCore                           0x00007fff46f2038e -[UIViewController _setPresentationController:] + 100
	20  UIKitCore                           0x00007fff46f18b9a -[UIViewController _presentViewController:modalSourceViewController:presentationController:animationController:interactionController:completion:] + 1256
	21  UIKitCore                           0x00007fff46f1a602 -[UIViewController _presentViewController:withAnimationController:completion:] + 4325
	22  UIKitCore                           0x00007fff46f1ce7b __63-[UIViewController _presentViewController:animated:completion:]_block_invoke + 98
	23  UIKitCore                           0x00007fff46f1d393 -[UIViewController _performCoordinatedPresentOrDismiss:animated:] + 511
	24  UIKitCore                           0x00007fff46f1cdd9 -[UIViewController _presentViewController:animated:completion:] + 187
	25  UIKitCore                           0x00007fff46f1d040 -[UIViewController presentViewController:animated:completion:] + 150
	26  Circle                              0x0000000108151cdb -[FIRIAMDisplayExecutor displayMessageLoadError:] + 407
	27  Circle                              0x0000000108151758 -[FIRIAMDisplayExecutor displayErrorForMessage:error:] + 509
	28  Circle                              0x000000010815337f __55-[FIRIAMDisplayExecutor displayForMessage:triggerType:]_block_invoke + 164
	29  Circle                              0x00000001081598eb __63-[FIRIAMMessageContentDataWithImageURL loadImageDataWithBlock:]_block_invoke.21 + 158
	30  Circle                              0x000000010815a17c __68-[FIRIAMMessageContentDataWithImageURL fetchImageFromURL:withBlock:]_block_invoke + 1258
	31  CFNetwork                           0x00007fff22d05a26 CFNetwork + 35366
	32  CFNetwork                           0x00007fff22d1887a _CFHTTPMessageSetResponseProxyURL + 16819
	33  Foundation                          0x00007fff2567de0f __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
	34  Foundation                          0x00007fff2567dd07 -[NSBlockOperation main] + 98
	35  Foundation                          0x00007fff25680c94 __NSOPERATION_IS_INVOKING_MAIN__ + 17
	36  Foundation                          0x00007fff2567cf07 -[NSOperation start] + 731
	37  Foundation                          0x00007fff256815ee __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 17
	38  Foundation                          0x00007fff256810f8 __NSOQSchedule_f + 182
	39  libdispatch.dylib                   0x000000010c0f6407 _dispatch_block_async_invoke2 + 83
	40  libdispatch.dylib                   0x000000010c0e7d6c _dispatch_client_callout + 8
	41  libdispatch.dylib                   0x000000010c0ea6de _dispatch_continuation_pop + 552
	42  libdispatch.dylib                   0x000000010c0e9ae9 _dispatch_async_redirect_invoke + 849
	43  libdispatch.dylib                   0x000000010c0f92a2 _dispatch_root_queue_drain + 351
	44  libdispatch.dylib                   0x000000010c0f9bac _dispatch_worker_thread2 + 132
	45  libsystem_pthread.dylib             0x00007fff518d171d _pthread_wqthread + 220
	46  libsystem_pthread.dylib             0x00007fff518d15c3 start_wqthread + 15
)
2019-10-20 00:44:26.904184+0800 Circle[57637:1636868] [Assert] Unsupported use of UIKit view-customization API off the main thread. -setHasDimmingView: sent to <_UIAlertControllerView: 0x7fa42553b440; frame = (0 0; 375 812); layer = <CALayer: 0x60000296f7a0>>
2019-10-20 00:44:26.904363+0800 Circle[57637:1636868] [Assert] Unsupported use of UIKit view-customization API off the main thread. -setShouldHaveBackdropView: sent to <_UIAlertControllerView: 0x7fa42553b440; frame = (0 0; 375 812); layer = <CALayer: 0x60000296f7a0>>
2019-10-20 00:44:26.904528+0800 Circle[57637:1636868] [Assert] Cannot be called with asCopy = NO on non-main thread.
2019-10-20 00:44:26.904677+0800 Circle[57637:1636868] [Assert] Unsupported use of UIKit view-customization API off the main thread. -setAlignsToKeyboard: sent to <_UIAlertControllerView: 0x7fa42553b440; frame = (0 0; 375 812); layer = <CALayer: 0x60000296f7a0>>
2019-10-20 00:44:26.910635+0800 Circle[57637:1636868] This application is modifying the autolayout engine from a background thread after the engine was accessed from the main thread. This can lead to engine corruption and weird crashes.
 Stack:(
	0   Foundation                          0x00007fff25802dbf _AssertAutolayoutOnAllowedThreadsOnly + 175
	1   Foundation                          0x00007fff25803500 -[NSISEngine withBehaviors:performModifications:] + 25
	2   UIKitCore                           0x00007fff4796d3a8 -[UIView(UIConstraintBasedLayout) _calculatedSystemLayoutSizeFittingSize:withHorizontalFittingPriority:verticalFittingPriority:hasIntentionallyCollapsedHeight:] + 1198
	3   UIKitCore                           0x00007fff4796e839 -[UIView(AdditionalLayoutSupport) _systemLayoutSizeFittingSize:withHorizontalFittingPriority:verticalFittingPriority:hasIntentionallyCollapsedHeight:] + 694
	4   UIKitCore                           0x00007fff46b4aec4 -[_UIAlertControllerView _minimumSizeForAllActions] + 593
	5   UIKitCore                           0x00007fff46b464a2 -[_UIAlertControllerView _itemSizeForHorizontalLayout:visualStyleRequiresActionRepresentationToFitItemSize:] + 36
	6   UIKitCore                           0x00007fff46b4672f -[_UIAlertControllerView _canLayOutActionsHorizontally] + 95
	7   UIKitCore                           0x00007fff46b469f5 -[_UIAlertControllerView _actionLayoutDirectionChanged] + 51
	8   UIKitCore                           0x00007fff46b4b686 -[_UIAlertControllerView _updateStyleForIdiomChange:] + 224
	9   UIKitCore                           0x00007fff46b4b509 -[_UIAlertControllerView _setVisualStyle:] + 282
	10  UIKitCore                           0x00007fff46b2c6ae -[UIAlertController loadView] + 397
	11  UIKitCore                           0x00007fff46f0895f -[UIViewController loadViewIfRequired] + 172
	12  UIKitCore                           0x00007fff46f0910c -[UIViewController view] + 27
	13  UIKitCore                           0x00007fff46f2038e -[UIViewController _setPresentationController:] + 100
	14  UIKitCore                           0x00007fff46f18b9a -[UIViewController _presentViewController:modalSourceViewController:presentationController:animationController:interactionController:completion:] + 1256
	15  UIKitCore                           0x00007fff46f1a602 -[UIViewController _presentViewController:withAnimationController:completion:] + 4325
	16  UIKitCore                           0x00007fff46f1ce7b __63-[UIViewController _presentViewController:animated:completion:]_block_invoke + 98
	17  UIKitCore                           0x00007fff46f1d393 -[UIViewController _performCoordinatedPresentOrDismiss:animated:] + 511
	18  UIKitCore                           0x00007fff46f1cdd9 -[UIViewController _presentViewController:animated:completion:] + 187
	19  UIKitCore                           0x00007fff46f1d040 -[UIViewController presentViewController:animated:completion:] + 150
	20  Circle                              0x0000000108151cdb -[FIRIAMDisplayExecutor displayMessageLoadError:] + 407
	21  Circle                              0x0000000108151758 -[FIRIAMDisplayExecutor displayErrorForMessage:error:] + 509
	22  Circle                              0x000000010815337f __55-[FIRIAMDisplayExecutor displayForMessage:triggerType:]_block_invoke + 164
	23  Circle                              0x00000001081598eb __63-[FIRIAMMessageContentDataWithImageURL loadImageDataWithBlock:]_block_invoke.21 + 158
	24  Circle                              0x000000010815a17c __68-[FIRIAMMessageContentDataWithImageURL fetchImageFromURL:withBlock:]_block_invoke + 1258
	25  CFNetwork                           0x00007fff22d05a26 CFNetwork + 35366
	26  CFNetwork                           0x00007fff22d1887a _CFHTTPMessageSetResponseProxyURL + 16819
	27  Foundation                          0x00007fff2567de0f __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
	28  Foundation                          0x00007fff2567dd07 -[NSBlockOperation main] + 98
	29  Foundation                          0x00007fff25680c94 __NSOPERATION_IS_INVOKING_MAIN__ + 17
	30  Foundation                          0x00007fff2567cf07 -[NSOperation start] + 731
	31  Foundation                          0x00007fff256815ee __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 17
	32  Foundation                          0x00007fff256810f8 __NSOQSchedule_f + 182
	33  libdispatch.dylib                   0x000000010c0f6407 _dispatch_block_async_invoke2 + 83
	34  libdispatch.dylib                   0x000000010c0e7d6c _dispatch_client_callout + 8
	35  libdispatch.dylib                   0x000000010c0ea6de _dispatch_continuation_pop + 552
	36  libdispatch.dylib                   0x000000010c0e9ae9 _dispatch_async_redirect_invoke + 849
	37  libdispatch.dylib                   0x000000010c0f92a2 _dispatch_root_queue_drain + 351
	38  libdispatch.dylib                   0x000000010c0f9bac _dispatch_worker_thread2 + 132
	39  libsystem_pthread.dylib             0x00007fff518d171d _pthread_wqthread + 220
	40  libsystem_pthread.dylib             0x00007fff518d15c3 start_wqthread + 15
)
2019-10-20 00:44:26.916928+0800 Circle[57637:1636868] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Modifications to the layout engine must not be performed from a background thread after it has been accessed from the main thread.'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23baa1ee __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff50864b20 objc_exception_throw + 48
	2   Foundation                          0x00007fff25802e3c -[NSISEngine tryToOptimizeReturningMutuallyExclusiveConstraints] + 0
	3   Foundation                          0x00007fff25803500 -[NSISEngine withBehaviors:performModifications:] + 25
	4   UIKitCore                           0x00007fff4796d3a8 -[UIView(UIConstraintBasedLayout) _calculatedSystemLayoutSizeFittingSize:withHorizontalFittingPriority:verticalFittingPriority:hasIntentionallyCollapsedHeight:] + 1198
	5   UIKitCore                           0x00007fff4796e839 -[UIView(AdditionalLayoutSupport) _systemLayoutSizeFittingSize:withHorizontalFittingPriority:verticalFittingPriority:hasIntentionallyCollapsedHeight:] + 694
	6   UIKitCore                           0x00007fff46b4aec4 -[_UIAlertControllerView _minimumSizeForAllActions] + 593
	7   UIKitCore                           0x00007fff46b464a2 -[_UIAlertControllerView _itemSizeForHorizontalLayout:visualStyleRequiresActionRepresentationToFitItemSize:] + 36
	8   UIKitCore                           0x00007fff46b4672f -[_UIAlertControllerView _canLayOutActionsHorizontally] + 95
	9   UIKitCore                           0x00007fff46b469f5 -[_UIAlertControllerView _actionLayoutDirectionChanged] + 51
	10  UIKitCore                           0x00007fff46b4b686 -[_UIAlertControllerView _updateStyleForIdiomChange:] + 224
	11  UIKitCore                           0x00007fff46b4b509 -[_UIAlertControllerView _setVisualStyle:] + 282
	12  UIKitCore                           0x00007fff46b2c6ae -[UIAlertController loadView] + 397
	13  UIKitCore                           0x00007fff46f0895f -[UIViewController loadViewIfRequired] + 172
	14  UIKitCore                           0x00007fff46f0910c -[UIViewController view] + 27
	15  UIKitCore                           0x00007fff46f2038e -[UIViewController _setPresentationController:] + 100
	16  UIKitCore                           0x00007fff46f18b9a -[UIViewController _presentViewController:modalSourceViewController:presentationController:animationController:interactionController:completion:] + 1256
	17  UIKitCore                           0x00007fff46f1a602 -[UIViewController _presentViewController:withAnimationController:completion:] + 4325
	18  UIKitCore                           0x00007fff46f1ce7b __63-[UIViewController _presentViewController:animated:completion:]_block_invoke + 98
	19  UIKitCore                           0x00007fff46f1d393 -[UIViewController _performCoordinatedPresentOrDismiss:animated:] + 511
	20  UIKitCore                           0x00007fff46f1cdd9 -[UIViewController _presentViewController:animated:completion:] + 187
	21  UIKitCore                           0x00007fff46f1d040 -[UIViewController presentViewController:animated:completion:] + 150
	22  Circle                              0x0000000108151cdb -[FIRIAMDisplayExecutor displayMessageLoadError:] + 407
	23  Circle                              0x0000000108151758 -[FIRIAMDisplayExecutor displayErrorForMessage:error:] + 509
	24  Circle                              0x000000010815337f __55-[FIRIAMDisplayExecutor displayForMessage:triggerType:]_block_invoke + 164
	25  Circle                              0x00000001081598eb __63-[FIRIAMMessageContentDataWithImageURL loadImageDataWithBlock:]_block_invoke.21 + 158
	26  Circle                              0x000000010815a17c __68-[FIRIAMMessageContentDataWithImageURL fetchImageFromURL:withBlock:]_block_invoke + 1258
	27  CFNetwork                           0x00007fff22d05a26 CFNetwork + 35366
	28  CFNetwork                           0x00007fff22d1887a _CFHTTPMessageSetResponseProxyURL + 16819
	29  Foundation                          0x00007fff2567de0f __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
	30  Foundation                          0x00007fff2567dd07 -[NSBlockOperation main] + 98
	31  Foundation                          0x00007fff25680c94 __NSOPERATION_IS_INVOKING_MAIN__ + 17
	32  Foundation                          0x00007fff2567cf07 -[NSOperation start] + 731
	33  Foundation                          0x00007fff256815ee __NSOPERATIONQUEUE_IS_STARTING_AN_OPERATION__ + 17
	34  Foundation                          0x00007fff256810f8 __NSOQSchedule_f + 182
	35  libdispatch.dylib                   0x000000010c0f6407 _dispatch_block_async_invoke2 + 83
	36  libdispatch.dylib                   0x000000010c0e7d6c _dispatch_client_callout + 8
	37  libdispatch.dylib                   0x000000010c0ea6de _dispatch_continuation_pop + 552
	38  libdispatch.dylib                   0x000000010c0e9ae9 _dispatch_async_redirect_invoke + 849
	39  libdispatch.dylib                   0x000000010c0f92a2 _dispatch_root_queue_drain + 351
	40  libdispatch.dylib                   0x000000010c0f9bac _dispatch_worker_thread2 + 132
	41  libsystem_pthread.dylib             0x00007fff518d171d _pthread_wqthread + 220
	42  libsystem_pthread.dylib             0x00007fff518d15c3 start_wqthread + 15
)

```

**Comment by ryanwilson** [comment]
Thanks for the report! Looks like the responsible code is here:

https://github.com/firebase/firebase-ios-sdk/blob/16fbcf6116a13844fb8c3147d88fd8d90a5115c9/Firebase/InAppMessaging/Flows/FIRIAMDisplayExecutor.m#L315

@christibbs it's also worth noting that `keyWindow` was deprecated in iOS 13 and we may need to fetch the window from the `connectedScenes` property.

**Comment by ardalahmet** [comment]
I'm experiencing the exact same crash. Here are my dependencies:

```
  - FirebaseInAppMessaging (0.15.5):
  - FirebaseInAppMessagingDisplay (0.15.5):
  - FirebaseMessaging (4.1.9):
```

I'm building with `Xcode 11.0 (11A420a)`.
My test device is an `iPhone X` and runs on `iOS 13.2`.

Is there any progress on this issue?
Thanks.

**Comment by christibbs** [comment]
Hey all, shooting to get a fix for this in this first release of December (the 9th).

**Comment by ardalahmet** [comment]
@christibbs Cool! Looking forward to it.

**Comment by ardalahmet** [comment]
@christibbs Is this fixed? If so, is it released? Thanks.

**Comment by cblaze22** [comment]
Is this fixed?

**Comment by paulb777** [comment]
The fix came in #4446 and will be released with the next milestone. You can track the the 6.14.0 milestone associated with this issue that will be closed with the next release.
